src/module/module.jl

```diff
@@ -407,6 +408,21 @@ function prune_with_map(I::smodule{spoly{T}}) where T <: Nemo.FieldElem
    return smodule{spoly{T}}(R, r),smatrix{spoly{T}}(R,TT_ptr)
 end
 
+@doc raw"""
+    prune_with_map_projection(id::smodule{spoly{T}}) where T <: Nemo.FieldElem
+
+Returns the module R minimally embedded in a free module such that the
+corresponding factor modules are isomorphic,
```

Comment:
I don't understand this docstring: There is one argument `id`; it is not stated how it related to `R`. And then the next line, is that transformation matrix meant to be another return value?

How about this (assuming I guessed the meaning right, which I probably didn't):
```suggestion
Return a module `R`, a transformation matrix `M` and a projection map `p`
satisfying the following properties: `R` is minimally embedded in a free
module `parent(R)` such that the quotient `parent(R)/R` is isomorphic to `id`,
with `M` describing this isomorphism, and `p` mapping `gen(id, i)` to
`gen(id, p(i))` 
```
I am still rather confused by the usage of `gen(i)` (what is `i`? An index ranging over `1 .. ngens(id)` maybe?!?), and why `p[i]` -- is `p` a vector, or a function, or what?